core/services/gateway/handlers/functions/allowlist.go

```diff
@@ -172,17 +186,44 @@ func (a *onchainAllowlist) updateFromContractV1(ctx context.Context, blockNum *b
 	if err != nil {
 		return errors.Wrap(err, "unexpected error during functions_allow_list.NewTermsOfServiceAllowList")
 	}
-	addrList, err := tosContract.GetAllAllowedSenders(&bind.CallOpts{
+
+	count, err := tosContract.GetAllowedSendersCount(&bind.CallOpts{
 		Pending:     false,
 		BlockNumber: blockNum,
 		Context:     ctx,
 	})
 	if err != nil {
-		return errors.Wrap(err, "error calling GetAllAllowedSenders")
+		return errors.Wrap(err, "unexpected error during functions_allow_list.GetAllowedSendersCount")
+	}
+
+	allowedSenderList := make([]common.Address, 0)
+	for idStart := uint64(0); idStart < count; idStart += uint64(a.config.CacheBatchSize) {
```

Comment:
Could we add a comment line here that mentions something like:
If the on-chain allow list is modified during this loop is running the change may not be included. It will be seen on the next query.